File: ip-messaging/users/token-generation/token-generation.m
// Initialize Chat Client
NSString *identifierForVendor = [[[UIDevice currentDevice] identifierForVendor] UUIDString];
NSString *tokenEndpoint = @"http://localhost:8000/token.php?device=%@";
NSString *urlString = [NSString stringWithFormat:tokenEndpoint, identifierForVendor];

// Make JSON request to server
NSURL *url = [NSURL URLWithString:urlString];
NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]];
NSURLSessionDataTask *dataTask = [session dataTaskWithURL:url completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
  if (data) {
    NSError *jsonError;
    NSDictionary *tokenResponse = [NSJSONSerialization JSONObjectWithData:data
                                                                  options:kNilOptions
                                                                    error:&jsonError];
    // Handle response from server
    if (!jsonError) {
      self.identity = tokenResponse[@"identity"];
      self.token = tokenResponse[@"token"]  
    } else {
      NSLog(@"error parsing token from server");
    }
  } else {
    NSLog(@"error fetching token from server");
  }
}];
[dataTask resume];

File: ip-messaging/users/token-generation/token-generation.swift
let deviceId = UIDevice.current.identifierForVendor!.uuidString
let urlString = "http://localhost:8000/token.php?device=\(deviceId)"

// Make HTTP request
if let requestURL = URL(string: urlString) {
  let session = URLSession(configuration: URLSessionConfiguration.default)
  let task = session.dataTask(with: requestURL, completionHandler: { (data, response, error) in
      if let data = data {
          do {
              let json = try JSONSerialization.jsonObject(with: data, options: []) as! [String:String]
              let token = json["token"]
              let identity = json["identity"]
          }
          catch let error as NSError {
            print ("Error with json, error = \(error)")
          }
          
      } else {
        print ("Error fetching json, error = \(error)")
      }
  })
  task.resume()
}